internal/auth/device_authorization.go

```diff
@@ -0,0 +1,179 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
```

Comment:
Borrowed the OAuth2 built in types for some marshaling, but couldn't use it for token exchange here as MS requires a custom scope to be passed, and this library doesn't support doing that when using refresh tokens